File: vps-renew-notification/src/payload/vps.rs
use serde::Deserialize;
use validator::Validate;

#[derive(Deserialize)]
pub enum ListVpsSort {
    #[serde(rename = "name")]
    Name,

    #[serde(rename = "name_desc")]
    NameDesc,

    #[serde(rename = "id")]
    ID,

    #[serde(rename = "id_desc")]
    IDDesc,
}

#[derive(Deserialize)]
pub struct ListVps {
    pub name: Option<String>,
    pub provider_id: Option<String>,
    pub sort: Option<ListVpsSort>,
}

#[derive(Deserialize, Validate)]
pub struct AddVps {
    #[validate(length(min = 20, max = 20, message = "请输入正确的ID"))]
    pub provider_id: String,

    #[validate(length(min = 1, max = 50, message = "请输入正确的名称"))]
    pub name: String,

    pub expire: String,
}
#[derive(Deserialize, Validate)]
pub struct EditVps {
    #[validate(length(min = 20, max = 20, message = "请输入正确的ID"))]
    pub id: String,

    #[validate(length(min = 20, max = 20, message = "请输入正确的ID"))]
    pub provider_id: String,

    #[validate(length(min = 1, max = 50, message = "请输入正确的名称"))]
    pub name: String,

    pub expire: String,
}

File: vps-renew-notification-go/web/payload/vps.go
package payload

import "time"

type AddVps struct {
	Name       string    `json:"name" form:"name" binding:"required,max=50"`
	ProviderID string    `json:"provider_id" form:"provider_id" binding:"required,len=20"`
	Expire     time.Time `json:"expire" form:"expire" binding:"required"`
}

type EditVps struct {
	ID string `json:"id" form:"id" binding:"required,len=20"`
	AddVps
}

type ListVpsSort = string

const (
	ListVpsSortName       ListVpsSort = "name"
	ListVpsSortNameDesc   ListVpsSort = "name_desc"
	ListVpsSortID         ListVpsSort = "id"
	ListVpsSortIDDesc     ListVpsSort = "id_desc"
	ListVpsSortExpire     ListVpsSort = "expire"
	ListVpsSortExpireDesc ListVpsSort = "expire_desc"
)

type ListVps struct {
	Name       string      `json:"name" form:"name"`
	ProviderID string      `json:"provider_id" form:"provider_id"`
	Sort       ListVpsSort `json:"sort" form:"sort"`
}
